internal/errors/errors.go

```diff
@@ -135,3 +136,39 @@ func IsErrMissingRequiredSecret(err error) bool {
 	_, ok := err.(ErrMissingRequiredSecret)
 	return ok
 }
+
+type ErrInvalidConfiguration struct {
+	message string
+}
+
+func NewErrInvalidConfiguration(cause error) ErrInvalidConfiguration {
+	return ErrInvalidConfiguration{message: cause.Error()}
+}
+
+func (e ErrInvalidConfiguration) Error() string {
+	return fmt.Sprintf("invalid configuration: %s", e.message)
+}
+
+// IsErrInvalidConfiguration: Reflect: returns true if the error is a ErrInvalidConfiguration
+func IsErrInvalidConfiguration(err error) bool {
+	_, ok := err.(ErrInvalidConfiguration)
+	return ok
+}
```

Comment:
I don't think we want this function.

Callers should be using:

```go
var invalidConfig ErrInvalidConfiguration
isInvalidConfig := errors.As(err, &invalidConfig)
```

Or, we should implement `func (e ErrInvalidConfiguration) Is(target error) bool`, letting callers do `errors.Is(err, ErrInvalidConfiguration{})`

Reply:
I assume the reasoning is that the error is readily available after the cast at the call site (rather than discarding it within this kind of function)?

I don't mind adjusting it here, no strong preference from me. I will again highlight that the other errors defined here used this approach, so we'll be going against the grain.

Reply:
The goal is to play nice with the `errors.Is`/`errors.As`, since that's the convention the go ecosystem has ended up on, and the best we've got.

It goes with the grain of the go stdlib, but against the grain of this file, which I think didn't realize these broader go conventions existed.